src/wam2layers/tracking/core.py

```diff
@@ -5,27 +5,12 @@
 from wam2layers.config import Config
 from wam2layers.utils.profiling import ProgressTracker
 
-
-def pad_boundaries(*args, periodic=False):
-    """Add boundary padding to all input arrays.
-
-    Arguments:
-        *args: Input arrays to which boundary padding will be added.
-        periodic: If True, apply periodic boundary conditions when padding in
-            the x-direction. If False, pad with zeros in the x and y directions.
-
-    Returns:
-        List of input arrays with boundary padding added.
-    """
-    zero_xy = partial(np.pad, pad_width=1, mode="constant", constant_values=0)
-    periodic_x = partial(np.pad, pad_width=((0, 0), (1, 1)), mode="wrap")
-    zero_y = partial(
-        np.pad, pad_width=((1, 1), (0, 0)), mode="constant", constant_values=0
-    )
-
-    if periodic:
-        return [periodic_x(zero_y(arg)) for arg in args]
-    return [zero_xy(arg) for arg in args]
+# Functions to pad poundaries of arrays
+pad_x_wrap = partial(np.pad, pad_width=((0, 0), (1, 1)), mode="wrap")
+pad_y_zero = partial(
+    np.pad, pad_width=((1, 1), (0, 0)), mode="constant", constant_values=0
+)
+pad_xy_zero = partial(np.pad, pad_width=1, mode="constant", constant_values=0)
 
 
 def horizontal_advection(s, u, v, periodic_x=False) -> np.ndarray:
```

Comment:
would be good to add a bit more inline comments in this function as these things are not straightforward of why what is happening, which would be very helpful if somebody revisits this code in a couple of years.

Reply:
Done